src/main/java/io/tackle/operator/TackleController.java

```diff
@@ -26,23 +36,34 @@
 
     @Override
     public DeleteControl deleteResource(Tackle tackle, Context<Tackle> context) {
-        String namespace = tackle.getMetadata().getNamespace();
-        kubernetesClient.customResources(Microservice.class).inNamespace(namespace).delete(kubernetesClient.customResources(Microservice.class).inNamespace(namespace).list().getItems());
-        kubernetesClient.customResources(PostgreSQL.class).inNamespace(namespace).delete(kubernetesClient.customResources(PostgreSQL.class).inNamespace(namespace).list().getItems());
-        kubernetesClient.customResources(Keycloak.class).inNamespace(namespace).delete(kubernetesClient.customResources(Keycloak.class).inNamespace(namespace).list().getItems());
-        kubernetesClient.customResources(Rest.class).inNamespace(namespace).delete(kubernetesClient.customResources(Rest.class).inNamespace(namespace).list().getItems());
-        kubernetesClient.customResources(Ui.class).inNamespace(namespace).delete(kubernetesClient.customResources(Ui.class).inNamespace(namespace).list().getItems());
-        Resource<Secret> dockerhubSecret = kubernetesClient.secrets().inNamespace(namespace).withName(AbstractController.DOCKERHUB_IMAGE_PULLER_SECRET_NAME);
-        if (dockerhubSecret.get() != null) {
-            dockerhubSecret.delete();
-            log.infof("Deleted Secret '%s' in namespace '%s'", AbstractController.DOCKERHUB_IMAGE_PULLER_SECRET_NAME, namespace);
-        }
+        final String namespace = tackle.getMetadata().getNamespace();
+        final String name = tackle.getMetadata().getName();
+        log.infof("Execution deleteResource for Tackle '%s' in namespace '%s'", name, namespace);
         return DeleteControl.DEFAULT_DELETE;
     }
 
     @Override
     public UpdateControl<Tackle> createOrUpdateResource(Tackle tackle, Context<Tackle> context) {
-        String namespace = tackle.getMetadata().getNamespace();
+        final String namespace = tackle.getMetadata().getNamespace();
+        final String name = tackle.getMetadata().getName();
+
+        BasicStatus status = tackle.getStatus();
+        if (status != null && status.getConditions()
+                .stream()
+                .anyMatch(condition ->
+                        CONDITION_TYPE_READY.equals(condition.getType()) &&
+                        CONDITION_STATUS_TRUE.equals(condition.getStatus()))) {
+                log.infof("Tackle '%s' CR already created, nothing to do.", tackle.getMetadata().getName());
+                return UpdateControl.noUpdate();
+        }
+
+        final OwnerReference tackleOwnerReference = new OwnerReferenceBuilder()
```

Comment:
From here it seems we are considering ONLY that creation messages arrive to this method, but it could be an update on a metadata or the single property Tackle has, and it should not imply creating or updating all the operands ( that could mean overriding information changed by the user : replicas, metadata, etc )